PORTAL/jobs/_pyperformance.py

```diff
@@ -647,16 +745,20 @@ def from_raw(cls, raw, *, fail=None):
             return None
 
     @classmethod
-    def _parse_value(cls, valuestr):
+    def _parse_value(cls, valuestr: str) -> _utils.ElapsedTimeWithUnits:
         return _utils.ElapsedTimeWithUnits.parse(valuestr, fail=True)
 
-    def __init__(self, source, byname=None):
+    def __init__(
+            self,
+            source: Any,
+            byname: Optional[Dict[str, str]] = None
+    ):
         _utils.check_str(source, 'source', required=True, fail=True)
         if not os.path.isabs(source):
             raise ValueError(f'expected an absolute source, got {source!r}')
         # XXX Further validate source as a filename?
 
-        _byname = {}
+        _byname: Dict[str, _utils.ElapsedTimeWithUnits] = {}
```

Comment:
The type here depends on the return type of `cls._parse_value()`.  So for `PyperfComparison` the type is actually `Dict[str, PyperfComparisonValue]`.